src/models/subscriptiondata.cpp

```diff
@@ -50,7 +50,7 @@ bool SubscriptionData::fromSettings() {
 
   logger.debug() << "Reading the subscription data from settings";
 
-  const QByteArray& json = settingsHolder->devices();
+  const QByteArray& json = settingsHolder->subscriptionData();
```

Comment:
Nice catch! Do you know what the symptoms of this bug was?

Reply:
Whenever we run `TaskGetSubscriptionDetails`, we compare the incoming json with what we have stored in settings before parsing it. Here we were always comparing new subscription data json with device settings json so they were always different and we would always have to re-parse (emitting `subscriptionDataChanged`). Don't know of any bugs this caused other than wasteful signal emission/binding reevaluations